### DISCUSSION TITLE
prevent layout from closing when input submitted

### DISCUSSION THREAD

**Discussion by mboros1** [MunifTanjim/nui.nvim#378] [Q&A]
I'm trying to use the input box to send commands, which will send the inputs to an external program updating other popups
<img width="1464" alt="image" src="https://github.com/MunifTanjim/nui.nvim/assets/13160711/62515b38-50e4-4e02-915d-8b38083f5907">
but when I submit, all of the boxes in the layout just close.

I can call a function in on submit to re-open the layout, but it causes the screen to flicker as it closes and re-opens which is a little annoying. Do you have advice on the best way to just have the input box clear the input text without closing when the on_submit function runs?

**Comment by MunifTanjim** [comment]
You can just do something like this:

```lua
  local prompt = "> "

  local input = Input({
    relative = "cursor",
    position = { row = 1, col = 0 },
    size = { width = 20 },
    border = "single",
  }, {
    prompt = prompt,
  })

  input:mount()

  vim.keymap.set({ "n", "i" }, "<Enter>", function()
    local value = string.sub(vim.api.nvim_get_current_line(), vim.fn.strwidth(prompt) + 1)
    print(value)
  end, { buffer = input.bufnr })
```

This will override the default behavior of `<Enter>` key (which is to submit and close the input).
